From e5204c4c75e7882ebd3dc931f76590d35aabfb3f Mon Sep 17 00:00:00 2001
From: Frederic Crozat <fcrozat@suse.com>
Date: Thu, 21 Feb 2013 16:09:40 +0100
Subject: [PATCH 188/204] fstab,mount: detect rbind as bind mount

Correctly detect rbind mount option as bind mount.

Fixes https://bugzilla.novell.com/show_bug.cgi?id=804575.
(cherry picked from commit 3f8ee7918207d7128d4edbc20a1e81f4c6e1110e)
---
 src/core/mount.c                      | 6 ++++++
 src/fstab-generator/fstab-generator.c | 4 +++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/core/mount.c b/src/core/mount.c
index 14f4863..0b64a1f 100644
--- a/src/core/mount.c
+++ b/src/core/mount.c
@@ -320,6 +320,12 @@ static bool mount_is_bind(MountParameters *p) {
         if (p->fstype && streq(p->fstype, "bind"))
                 return true;
 
+        if (mount_test_option(p->options, "rbind"))
+                return true;
+
+        if (p->fstype && streq(p->fstype, "rbind"))
+                return true;
+
         return false;
 }
 
diff --git a/src/fstab-generator/fstab-generator.c b/src/fstab-generator/fstab-generator.c
index 23e5051..9b64462 100644
--- a/src/fstab-generator/fstab-generator.c
+++ b/src/fstab-generator/fstab-generator.c
@@ -199,7 +199,9 @@ static bool mount_is_bind(struct mntent *me) {
 
         return
                 hasmntopt(me, "bind") ||
-                streq(me->mnt_type, "bind");
+                streq(me->mnt_type, "bind") ||
+                hasmntopt(me, "rbind") ||
+                streq(me->mnt_type, "rbind");
 }
 
 static bool mount_is_network(struct mntent *me) {
-- 
1.8.1.5

